docs/development/data-flow/README.md

```diff
@@ -0,0 +1,98 @@
+# Data flow and Usage
+
+Regardless of the type of installation of OpenProject, the following diagram provides a high-level overview of through which systems data related to OpenProject is flowing.
+
+
+
+```mermaid
+%%{init: {'theme':'neutral'}}%%
+
+flowchart TD
+  A[Client Browser] -->|"HTTP(s) requests"| B(Load Balancer / Proxy)
+  A1[API or Native clients] -->|"HTTP(s) requests"| B(Load Balancer / Proxy)
+  B -->|Proxy| openproject
+
+  subgraph openproject[OpenProject]
+    C[Puma app server]
+    D[Background worker]
+  end
+    
+
+  subgraph integrations[External integrations]
+	  O[Other integrations]
+    N[Nextcloud]
+  end
+  
+  subgraph services[Services]
+  	M[memcached]
+	  P[PostgreSQL]
+	  S[Object storage or NFS]
+  end
+  
+
+  openproject <--> services
+  openproject --> integrations
+  B <--> integrations
+	
+```
+
+As a web application, the primary data flow is between the Client browser (or attached API clients) through an external proxying web server (a load balancer or proxying server). We're assuming it is the one responsible for terminating TLS connections for the course of this document - although encrypted connections between Load balancer and Puma server are possible.
```

Comment:
There might not always be a LB set up. Just mentioning it because it might confuse people who just want to have one machine providing OP.